cmake/modules/SwiftSharedCMakeConfig.cmake

```diff
@@ -162,10 +157,30 @@ macro(swift_common_standalone_build_config product is_cross_compiling)
 
   # Save and restore variables that LLVM configuration overrides.
   set(LLVM_TOOLS_BINARY_DIR_saved "${LLVM_TOOLS_BINARY_DIR}")
+
+  # If we already have a cached value for LLVM_ENABLE_ASSERTIONS, save the value.
   set(LLVM_ENABLE_ASSERTIONS_saved "${LLVM_ENABLE_ASSERTIONS}")
+  # Then we import LLVMCONFIG_FILE. This is going to override whatever cached
+  # value we have for LLVM_ENABLE_ASSERTIONS.
   include(${LLVMCONFIG_FILE})
   set(LLVM_TOOLS_BINARY_DIR "${LLVM_TOOLS_BINARY_DIR_saved}")
-  set(LLVM_ENABLE_ASSERTIONS "${LLVM_ENABLE_ASSERTIONS_saved}")
+
+  # If we did not have a cached value for LLVM_ENABLE_ASSERTIONS, set
+  # LLVM_ENABLE_ASSERTIONS_saved to be the ENABLE_ASSERTIONS value from LLVM so
+  # we follow LLVMConfig.cmake's value by default if nothing is provided.
+  if (NOT LLVM_ENABLE_ASSERTIONS_saved)
```

Comment:
This treats "assertions explicitly disabled" and "assertions not set" the same, doesn't it?